sql/core/src/test/scala/org/apache/spark/sql/ConfigBehaviorSuite.scala

```diff
@@ -39,9 +39,8 @@ class ConfigBehaviorSuite extends QueryTest with SharedSparkSession {
     def computeChiSquareTest(): Double = {
       val n = 10000
       // Trigger a sort
-      // Range has range partitioning in its output now. To have a range shuffle, we
-      // need to run a repartition first.
-      val data = spark.range(0, n, 1, 1).repartition(10).sort($"id".desc)
+      // Range has range partitioning in its output now.
+      val data = spark.range(0, n, 1, 10).sort($"id".desc)
```

Comment:
why this change?

Reply:
This test is failed because of pruning repartition, the partition number is not right in this case.

Reply:
The test wants to test a sort with shuffle. Do we still have shuffle in this query?

Reply:
Yes

Reply:
I'm a bit confused. `spark.range` reports `RangePartitioning`, so there shouldn't be any shuffles. What gets changed?

Reply:
```
== Optimized Logical Plan ==
Sort [id#8L DESC NULLS LAST], true
+- Range (0, 100, step=1, splits=Some(10))

== Physical Plan ==
*(2) Sort [id#8L DESC NULLS LAST], true, 0
+- Exchange rangepartitioning(id#8L DESC NULLS LAST, 4), true, [id=#33]
   +- *(1) Range (0, 100, step=1, splits=10)
```
The sort api will add global=true in SortExec, similar with orderby, which will cause SortExec's requiredChildDistribution to be parsed into OrderedDistribution.
EnsureRequirements will add a ShuffleExchangeExec with rangePartitioning after OrderedDistribution.

Reply:
In the current master
```
scala> spark.range(0, 10000, 1, 10).sort("id").explain(true)
== Parsed Logical Plan ==
'Sort ['id ASC NULLS FIRST], true
+- Range (0, 10000, step=1, splits=Some(10))

== Analyzed Logical Plan ==
id: bigint
Sort [id#8L ASC NULLS FIRST], true
+- Range (0, 10000, step=1, splits=Some(10))

== Optimized Logical Plan ==
Range (0, 10000, step=1, splits=Some(10))

== Physical Plan ==
*(1) Range (0, 10000, step=1, splits=10)
```

If we add a shuffle now, then it's a regression and we should fix.